mixer/pkg/adapter/handler.go

```diff
@@ -54,6 +56,15 @@ type (
 		Data []byte
 	}
 
+	// RemoteGenerateAttributesHandler calls remote APA adapter
+	RemoteGenerateAttributesHandler interface {
+		Handler
+
+		// HandleRemoteGenAttrs performs APA call based on pre encoded instances and returns the decoded output as an attribute bag.
+		// The caller must release the attribute bag on success.
+		HandleRemoteGenAttrs(ctx context.Context, encodedInstance *EncodedInstance, bag attribute.Bag) (*attribute.MutableBag, error)
```

Comment:
The return arg does not have to be a `*MutableBag`, just `attribute.Bag` will suffice. The caller will not modify the bag.

Alternatively it could be `ctx context.Context, encodedInstance *EncodedInstance, bag attribute.Bag, out *attribute.MutableBag) error `

Reply:
Done